src/client/dfuse/ops/setattr.c

```diff
@@ -16,6 +16,11 @@ dfuse_cb_setattr(fuse_req_t req, struct dfuse_inode_entry *ie,
 
 	DFUSE_TRA_DEBUG(ie, "flags %#x", to_set);
 
+	if (to_set & (FUSE_SET_ATTR_UID | FUSE_SET_ATTR_GID)) {
+		DFUSE_TRA_INFO(ie, "File uid/gid support not enabled");
+		D_GOTO(err, rc = ENOTSUP);
+	}
+
```

Comment:
This is interesting because some of the datamover tools (`dcp --preserve`) will try to copy the ownership. It's simple enough for `dcp` to just ignore `chown` for DFS. But a dfuse path is just seen as a POSIX path to `dcp`, so the `chown` call will always fail if a dfuse path is given as the destination to `dcp --preserve`.
Certainly not something that dfuse itself should have to worry about, but I wonder if there is a "conditional" chown that only runs if the owners are actually different? (Of course, `dcp` could always stat and check itself)

Reply:
The public behaviour of dfuse is unchanged by this, it'll still correctly return ENOTSUP and dcp should check for this.

I have a feature in-flight to add file ownership support, that should be in review as well - but it's layered on top of dfs using xattrs to store the data so will be an "opt-in" feature for containers.

Reply:
> return ENOTSUP and dcp should check for this

Right, I agree. It's more so that it's a **slight** incompatibility with standard POSIX. Anyway, it's not a fatal error in `dcp`, and I think we can easily handle it.

> a feature in-flight to add file ownership support

Nice!